### PR TITLE
Utils build

### PR DESCRIPTION
**Pull request by MagicMaxMark**
I finished the Utils code

### CODE REVIEW

File: Assets/Scripts/FoundationScript.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public void SetCardPositions()
         }
     }
 
-    public void Clicked()
+    public void Clicked(GameObject cardClciked)
```
Comment (SHELLEYJ32-zz) [inline Assets/Scripts/FoundationScript.cs:31]: 1. Spelling mistake "cardClciked"
2. Make sure UML reflect this change (currently the parameter is not reflected) and inform the other programmer who is initially working on this script.

File: Assets/Scripts/UtilsScript.cs [hunk 2]
```diff
@@ -22,36 +22,42 @@ void Awake()
...
         }
     }
 
 
     void Match()
     {
-        //clcikedCard.GetComponent<CardScript>().container = 
+        selectedCards[0].GetComponent<CardScript>().RemoveCard();
+        selectedCards[0].GetComponent<CardScript>().AddCard();
+        clcikedCard.GetComponent<CardScript>().RemoveCard();
```
Comment (SHELLEYJ32-zz) [inline Assets/Scripts/UtilsScript.cs:59]: 1. Spelling mistake "clcikedCard".
2. According to UML, this method should be public.
3. This method is not in the google doc for Base Reactor Code Implementation: https://docs.google.com/document/d/1yjgbBI4hG4TVPOF-W-Q_afr8eO5arjcxwDMsVYtM62o/edit?pli=1
4. This method should also interact with MatchedCardsScript by adding cards to the list if they match. Currently, no method in either script checks if the cards actually match.

File: Assets/Scripts/UtilsScript.cs [hunk 3]
```diff
@@ -63,51 +69,4 @@ void Update()
         }
```
Comment (SHELLEYJ32-zz) [inline Assets/Scripts/UtilsScript.cs:69]: this function is not in UML

**Review by SHELLEYJ32-zz — CHANGES_REQUESTED** [review]
See detailed comments

**Comment by SHELLEYJ32-zz** [comment]
This branch involves changes in other programmers's scripts. UtilsScript is also an important script all other scripts depend on, so making this branch and not informing other programmers through Slack is not a good practice. You can also make direct changes in the original branch FirstBuild.

File: Assets/Scripts/UtilsScript.cs (hunk 3 above)
Comment (MagicMaxMark) [reply Assets/Scripts/UtilsScript.cs:69]: added to UML

File: Assets/Scripts/UtilsScript.cs (hunk 2 above)
Comment (MagicMaxMark) [reply Assets/Scripts/UtilsScript.cs:59]: fixed in latest version

File: Assets/Scripts/FoundationScript.cs (hunk 1 above)
Comment (MagicMaxMark) [reply Assets/Scripts/FoundationScript.cs:31]: updated UML

**Pull request closed without merging**